// @aptos-labs/wallet-adapter-ant-design/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  WalletSelector: () => WalletSelector
});
module.exports = __toCommonJS(src_exports);

// src/WalletSelector.tsx
var import_react = require("react");
var import_antd = require("antd");
var import_wallet_adapter_react = require("@aptos-labs/wallet-adapter-react");

// src/utils.ts
var truncateAddress = (address) => {
  if (!address)
    return;
  return `${address.slice(0, 6)}...${address.slice(-5)}`;
};

// src/WalletSelector.tsx
var import_jsx_runtime = require("react/jsx-runtime");
var { Text } = import_antd.Typography;
function WalletSelector() {
  const [walletSelectorModalOpen, setWalletSelectorModalOpen] = (0, import_react.useState)(false);
  const { connect, disconnect, account, wallets, connected } = (0, import_wallet_adapter_react.useWallet)();
  const onWalletButtonClick = () => {
    if (connected) {
      disconnect();
    } else {
      setWalletSelectorModalOpen(true);
    }
  };
  const onWalletSelected = (wallet) => {
    connect(wallet);
    setWalletSelectorModalOpen(false);
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_jsx_runtime.Fragment, {
    children: [
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_antd.Button, {
        className: "wallet-button",
        onClick: () => onWalletButtonClick(),
        children: connected ? truncateAddress(account == null ? void 0 : account.address) : "Connect Wallet"
      }),
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_antd.Modal, {
        title: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
          className: "wallet-modal-title",
          children: "Connect Wallet"
        }),
        centered: true,
        open: walletSelectorModalOpen,
        onCancel: () => setWalletSelectorModalOpen(false),
        footer: [],
        closable: false,
        children: !connected && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_antd.Menu, {
          children: wallets.map((wallet) => {
            return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_antd.Menu.Item, {
              onClick: wallet.readyState === import_wallet_adapter_react.WalletReadyState.Installed ? () => onWalletSelected(wallet.name) : () => window.open(wallet.url),
              children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("div", {
                className: "wallet-menu-wrapper",
                children: [
                  /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("div", {
                    className: "wallet-name-wrapper",
                    children: [
                      /* @__PURE__ */ (0, import_jsx_runtime.jsx)("img", {
                        src: wallet.icon,
                        width: 25,
                        style: { marginRight: 10 }
                      }),
                      /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Text, {
                        className: "wallet-selector-text",
                        children: wallet.name
                      })
                    ]
                  }),
                  wallet.readyState === import_wallet_adapter_react.WalletReadyState.Installed ? /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_antd.Button, {
                    className: "wallet-connect-button",
                    children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Text, {
                      className: "wallet-connect-button-text",
                      children: "Connect"
                    })
                  }) : /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Text, {
                    className: "wallet-connect-install",
                    children: "Install"
                  })
                ]
              })
            }, wallet.name);
          })
        })
      })
    ]
  });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WalletSelector
});
